### ISSUE TITLE
[ixwebsocket] build failure

### ISSUE THREAD

**Issue by cnSchwarzer** [microsoft/vcpkg#15451]
**Host Environment**
- OS: Windows 10 20H2
- Compiler: VS 2019 Latest 

**To Reproduce**
`./vcpkg install ixwebsocket`
Create a project using the library
Build

**Failure logs** 
Translated:
Cannot resolve symbol (无法解析的外部符号) CertOpenStore...... etc..

**Additional context**
Add `#pragma comment(lib, "crypt32.lib")` to the source solves the issue.
Seems missing the dependency.

**Comment by JonLiu1993** [comment]
@cnSchwarzer ,Thanks for posting this issue,The port you need to use is xiwebsocket? I think what you need is ixwebsocket, did you write it wrong?

**Comment by cnSchwarzer** [comment]
> @cnSchwarzer ,Thanks for posting this issue,The port you need to use is xiwebsocket? I think what you need is ixwebsocket, did you write it wrong?

You're right, sorry for that ^_^

**Comment by JonLiu1993** [comment]
@cnSchwarzer ,I just reproduced on my local computer that this port does have a bug, but it is not the problem you reported. I changed the 30th line share/${port} of portfile.cmake to share/${PORT} and it can be used normally. Build, you can try

**Comment by JonLiu1993** [comment]
Rebuild started...
1>------ Rebuild All started: Project: ZERO_CHECK, Configuration: Debug Win32 ------
1>Checking Build System
2>------ Rebuild All started: Project: test, Configuration: Debug Win32 ------
2>Building Custom Rule C:/Users/vzhli17/source/repos/CMakeProject20/CMakeProject20/CMakeLists.txt
2>CMakeProject20.cpp
2>test.vcxproj -> C:\Users\vzhli17\source\repos\CMakeProject20\out\build\x86-Debug\CMakeProject20\Debug\test.exe
3>------ Skipped Rebuild All: Project: ALL_BUILD, Configuration: Debug Win32 ------
3>Project not selected to build for this solution configuration 
========== Rebuild All: 2 succeeded, 0 failed, 1 skipped ==========

**Comment by cnSchwarzer** [comment]
@JonLiu1993 
Hi, please try reproduce with following steps again, sorry. Since I am not using cmake but msbuild, I'm not sure if you successfully fixed.

1.`./vcpkg install ixwebsocket[ssl]:x64-windows`
2.Create a new "C++ Console Application (C++, Windows, Console)" project with VS2019. 
3.**Switch target to Debug, x64**
4.Replace `main.cpp` with following code.
```
#include <iostream>
#include <ixwebsocket/IXWebSocket.h>
#include <ixwebsocket/IXNetSystem.h>

int main()
{
	ix::initNetSystem();
	ix::WebSocket ws;
	ws.setUrl("wss://danmuproxy.douyu.com:8501"); 
}
```
5.Try Build the project, On my computer it failed with:
```
已启动生成…
1>------ 已启动生成: 项目: ixwebsocket_regression, 配置: Debug x64 ------
1>ixwebsocket_regression.cpp
1>ixwebsocket.lib(IXSocketMbedTLS.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_CertOpenStore，函数 "private: bool __cdecl ix::SocketMbedTLS::loadSystemCertificates(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?loadSystemCertificates@SocketMbedTLS@ix@@AEAA_NAEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) 中引用了该符号
1>ixwebsocket.lib(IXSocketMbedTLS.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_CertCloseStore，函数 "private: bool __cdecl ix::SocketMbedTLS::loadSystemCertificates(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?loadSystemCertificates@SocketMbedTLS@ix@@AEAA_NAEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) 中引用了该符号
1>ixwebsocket.lib(IXSocketMbedTLS.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_CertEnumCertificatesInStore，函数 "private: bool __cdecl ix::SocketMbedTLS::loadSystemCertificates(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?loadSystemCertificates@SocketMbedTLS@ix@@AEAA_NAEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) 中引用了该符号
1>ixwebsocket.lib(IXSocketMbedTLS.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_CertFreeCertificateContext，函数 "private: bool __cdecl ix::SocketMbedTLS::loadSystemCertificates(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?loadSystemCertificates@SocketMbedTLS@ix@@AEAA_NAEAV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z) 中引用了该符号
1>E:\Work\ixwebsocket_regression\x64\Debug\ixwebsocket_regression.exe : fatal error LNK1120: 4 个无法解析的外部命令
1>已完成生成项目“ixwebsocket_regression.vcxproj”的操作 - 失败。
========== 生成: 成功 0 个，失败 1 个，最新 0 个，跳过 0 个 ==========
```
6.If we add `#pragma comment (lib, "crypt32")` at first, it compiles.

7.If you edit `main.cpp` with this code:
```
#include <iostream>
#include <ixwebsocket/IXWebSocket.h>
#include <ixwebsocket/IXNetSystem.h>

int main()
{
	ix::initNetSystem(); 
}
```
8.It also failed with:
```
已启动生成…
1>------ 已启动生成: 项目: ixwebsocket_regression, 配置: Debug x64 ------
1>ixwebsocket_regression.cpp
1>ixwebsocket.lib(IXNetSystem.cpp.obj) : error LNK2019: 无法解析的外部符号 __WSAFDIsSet，函数 "int __cdecl ix::poll(struct pollfd *,unsigned long,int)" (?poll@ix@@YAHPEAUpollfd@@KH@Z) 中引用了该符号
1>ixwebsocket.lib(IXNetSystem.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_select，函数 "int __cdecl ix::poll(struct pollfd *,unsigned long,int)" (?poll@ix@@YAHPEAUpollfd@@KH@Z) 中引用了该符号
1>ixwebsocket.lib(IXNetSystem.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_WSAStartup，函数 "bool __cdecl ix::initNetSystem(void)" (?initNetSystem@ix@@YA_NXZ) 中引用了该符号
1>ixwebsocket.lib(IXNetSystem.cpp.obj) : error LNK2019: 无法解析的外部符号 __imp_WSACleanup，函数 "bool __cdecl ix::uninitNetSystem(void)" (?uninitNetSystem@ix@@YA_NXZ) 中引用了该符号
1>E:\Work\ixwebsocket_regression\x64\Debug\ixwebsocket_regression.exe : fatal error LNK1120: 4 个无法解析的外部命令
1>已完成生成项目“ixwebsocket_regression.vcxproj”的操作 - 失败。
========== 生成: 成功 0 个，失败 1 个，最新 0 个，跳过 0 个 ==========
``` 
9.If we add `#pragma comment (lib, "WS2_32")` at first, it compiles.

I have no idea why `ixwebsocket` not importing these libs automatically, is it because i am targeting x64? Please look into it, thanks! ^_^

**Comment by JonLiu1993** [comment]
@cnSchwarzer ,I confirmed your suggestion locally, it is feasible, it seems that there are two bugs in this port, I will solve them together.

**Comment by JackBoosY** [comment]
Ping for response.
Did you run command `./vcpkg integrate install` first?

**Comment by FrankXie05** [comment]
@cnSchwarzer Does the problem still exist? The fix PR has been merged #16710. If you are using MSBuild to build, make sure you run the command first `./vcpkg integrate install`. 
I will close this issue because of no reply for a long time, if there is progress please reply me and open it. :)
